GSoC-2025/application_template.md

```diff
@@ -1,87 +0,0 @@
-# Application template
```

Comment:
You shouldn't remove the template in the PR (so others can use it for other proposals)